sklearn/model_selection/_split.py

```diff
@@ -1770,90 +1759,55 @@ def split(self, X, y, groups=None):
-                                 (train_size + test_size))
-        elif np.asarray(train_size).dtype.kind != 'i':
-            # int values are checked during split based on the input
-            raise ValueError("Invalid value for train_size: %r" % train_size)
-
-
-def _validate_shuffle_split(n_samples, test_size, train_size):
+def _validate_shuffle_split(n_samples, test_size, train_size,
+                            default_test_size=None):
     """
     Validation helper to check if the test/test sizes are meaningful wrt to the
     size of the data (n_samples)
     """
-    if (test_size is not None and
-            (np.asarray(test_size).dtype.kind == 'i' and
-                (test_size >= n_samples or test_size <= 0)) or
-            (np.asarray(test_size).dtype.kind == 'f' and
-                (test_size <= 0 or test_size >= 1))):
-        raise ValueError('test_size=%d should be either positive and smaller '
-                         'than the number of samples %d or a float in the '
-                         '(0,1) range' % (test_size, n_samples))
-
-    if (train_size is not None and
-            (np.asarray(train_size).dtype.kind == 'i' and
-                (train_size >= n_samples or train_size <= 0)) or
-            (np.asarray(train_size).dtype.kind == 'f' and
-                (train_size <= 0 or train_size >= 1))):
-        raise ValueError('train_size=%d should be either positive and smaller '
-                         'than the number of samples %d or a float in the '
-                         '(0,1) range' % (train_size, n_samples))
-
-    if test_size == "default":
-        test_size = 0.1
-
-    if np.asarray(test_size).dtype.kind == 'f':
+    if test_size is None and train_size is None:
+        test_size = default_test_size
+
+    test_size_type = np.asarray(test_size).dtype.kind
+    train_size_type = np.asarray(train_size).dtype.kind
+
+    if (test_size_type == 'i' and (test_size >= n_samples or test_size <= 0)
+       or test_size_type == 'f' and (test_size <= 0 or test_size >= 1)):
+        raise ValueError('test_size={0} should be either positive and smaller'
+                         ' than the number of samples {1} or a float in the '
+                         '(0, 1) range'.format(test_size, n_samples))
+
+    if (train_size_type == 'i' and (train_size >= n_samples or train_size <= 0)
+       or train_size_type == 'f' and (train_size <= 0 or train_size >= 1)):
+        raise ValueError('train_size={0} should be either positive and smaller'
+                         ' than the number of samples {1} or a float in the '
+                         '(0, 1) range'.format(train_size, n_samples))
+
+    if train_size is not None and train_size_type not in ('i', 'f'):
+        raise ValueError("Invalid value for train_size: {}".format(train_size))
+    if test_size is not None and test_size_type not in ('i', 'f'):
+        raise ValueError("Invalid value for test_size: {}".format(test_size))
```

Comment:
Technically it should be a `TypeError` but I guess it's too late to change now as it was already raising a `ValueError` for this case in released versions of scikit-learn.

Reply:
yeah I guess so...